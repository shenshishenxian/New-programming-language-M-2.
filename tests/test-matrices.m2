int main() {
  matrix int [4][4] m33;
  m33 = [[2,9,1,1];
        [0,2,1,1];
        [1,1,1,1];
        [3,1,1,1]];
  myprint(m33:tr);
  return 0;
}

void myprint(matrix int [4][4] M){
  int i;
  int j;
  for(i = 0; i < M:rows; ++i){
    for(j = 0; j < M:cols; ++j){
      printInt(M[i,j]);
    }
    printStr("");
  }
}

