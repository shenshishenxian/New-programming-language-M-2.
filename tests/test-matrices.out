2	0	1	3	
9	2	1	1	
1	1	1	1	
1	1	1	1	
